src/typechecker.erl

```diff
@@ -854,6 +854,9 @@ int_range_to_types({neg_inf, I}) when I > 0 ->
     [type(neg_integer),
      {type, erl_anno:new(0), range, [{integer, erl_anno:new(0), 0}
                                     ,{integer, erl_anno:new(0), I}]}];
+int_range_to_types({neg_inf, I}) -> % I < -1
+    [{type, erl_anno:new(0), range, [{integer, erl_anno:new(0), neg_inf}
```

Comment:
The integer `neg_inf`, interesting! Perhaps add this to `gradualizer_type:abstract_type()`?

Reply:
Yeah, maybe that's a good idea actually. We need to be able to talk about ranges which contain negative and positive infinity in order to do accurate type refinement. I initially thought that we only needed these types during intermediate stages of type refinement but you're right that they will leak out into actual types. I'll ponder this some more but I think your suggestion makes sense.

Reply:
Yeah. We should have some way to "rebase" our version of the abstract type when it's updated in OTP as well... It's not good if we divert too much...

Reply:
I've now added `neg_inf` and `pos_inf` to `gradualizer_type:abstract_type()`.

Regarding keeping up with the OTP version of abstract type; sure, we should try to keep up. But how often does it actually change? Perhaps there is a relatively easy to make some kind of diff between our type and OTP and also different version of OTP and track the changes that way. Though I don't think this will be a huge problem in practice.